File: Educational Codeforces Round 138 (Rated for Div. 2)/f.cpp

#include <iostream>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;
        string s;
        cin >> s;
        int c = 0;
        int total = 0;
        bool cluster = false;
        for (int i = 0; i < n; i++)
        {
            if (s[i] == '1')
            {
                c = i + 1;
            }
            if (s[i] == '1' && !cluster)
            {
                total++;
                cluster = true;
            }
            else if (s[i] == '0')
            {
                cluster = false;
            }
        }

        if (c == n)
        {
            total--;
        }
        cout << total << endl;
    }
}

File: Educational Codeforces Round 138 (Rated for Div. 2)/f.py
t = int(input())
for _ in range(t):
	n = int(input())
	s = list(map(int,input()))
	cluster = False
	c = 0
	total = 0
	for j,i in enumerate(s):
		if i == 1:
			c = j + 1
		if i == 1 and not cluster:
			total+=1
			cluster = True
		elif i == 0:
			cluster = False
	if c == n:
		total -=1
	print(total)
